src/systems/sound-effects-system.js

```diff
@@ -0,0 +1,245 @@
+      this.soundFor.set("pen_undo_draw", buffer);
+      this.soundFor.set("pen_stop_draw", buffer);
+      this.soundFor.set("pen_change_radius", buffer);
+      this.soundFor.set("pen_change_color", buffer);
+      this.soundFor.set("toggle_mic", buffer);
+    });
+    getBuffer(TELEPORT_ARC, this.ctx).then(buffer => {
+      this.sounds.teleportArc = buffer;
+    });
+    getBuffer(QUICK_TURN, this.ctx).then(buffer => {
+      this.sounds.teleportEnd = buffer;
+    });
+    getBuffer(TAP_MELLOW, this.ctx).then(buffer => {
+      this.sounds.snapRotate = buffer;
+      this.soundFor.set("snap_rotate_left", buffer);
+      this.soundFor.set("snap_rotate_right", buffer);
+    });
+    getBuffer(PEN_SPAWN, this.ctx).then(buffer => {
+      this.sounds.spawnPen = buffer;
+      this.soundFor.set("spawn_pen", buffer);
+    });
+    getBuffer(PEN_DRAW, this.ctx).then(buffer => {
+      this.sounds.penStartDraw = buffer;
+      this.soundFor.set("pen_start_draw", buffer);
+    });
+    getBuffer(CAMERA_SNAPSHOT, this.ctx).then(buffer => {
+      this.sounds.cameraSnapshot = buffer;
+      this.soundFor.set("camera_tool_took_snapshot", buffer);
+    });
+    getBuffer(WELCOME, this.ctx).then(buffer => {
+      this.sounds.enterScene = buffer;
+      this.soundFor.set("enter_scene", buffer);
+    });
+    getBuffer(QUACK, this.ctx).then(buffer => {
+      this.sounds.quack = buffer;
+      this.soundFor.set("quack", buffer);
+    });
+    getBuffer(SPECIAL_QUACK, this.ctx).then(buffer => {
+      this.sounds.specialQuack = buffer;
+      this.soundFor.set("special_quack", buffer);
+    });
+    getBuffer(POP, this.ctx).then(buffer => {
+      this.sounds.chatMessage = buffer;
+      this.soundFor.set("chat_message", buffer);
+    });
+    getBuffer(FREEZE, this.ctx).then(buffer => {
+      this.sounds.freeze = buffer;
+      this.soundFor.set("freeze", buffer);
+    });
+    getBuffer(TACK, this.ctx).then(buffer => {
+      this.sounds.pin = buffer;
+      this.soundFor.set("pin", buffer);
+    });
+    getBuffer(MEDIA_LOADING, this.ctx).then(buffer => {
+      this.sounds.mediaLoading = buffer;
+    });
+    getBuffer(MEDIA_LOADED, this.ctx).then(buffer => {
+      this.sounds.mediaLoaded = buffer;
+      this.soundFor.set("media_loaded", buffer);
+    });
+
+    this.mediaLoadingSources = [];
+    this.onMediaLoading = () => {
+      this.mediaLoadingSources.push(playSoundLooped(this.sounds.mediaLoading, this.ctx));
+    };
+    this.onMediaLoaded = () => {
+      if (this.mediaLoadingSources.length) {
+        this.mediaLoadingSources.pop().stop();
+      }
+      this.pendingEffects.push("media_loaded");
+    };
+    scene.addEventListener("media-loaded", this.onMediaLoaded);
+    scene.addEventListener("media-loading", this.onMediaLoading);
+  }
+
+  cleanUp(scene) {
+    scene.removeEventListener("media-loaded", this.onMediaLoaded);
+    scene.removeEventListener("media-loading", this.onMediaLoading);
+  }
+
+  shouldTick() {
```

Comment:
I would ditch this and have a promise that resolve once all sounds are ready set `soundsAreReady`